### PR TITLE
feat: add support for HTTP proxies

### LINKED ISSUE #48: Proxy settings for NPM registry
I'm using a private registry that is only available with proxy. Is there a way to setup proxy URL?

### PR DESCRIPTION
**Pull request by fgreinacher**
This adds support for HTTP proxies configured in environment variables.

I think it's best to not have an option for this and do the right thing out of the box (like [cURL](https://everything.curl.dev/usingcurl/proxies/env.html), [axios](https://github.com/axios/axios/blob/v1.x/README.md?plain=1#L561-L564), [npm](https://docs.npmjs.com/cli/v6/using-npm/config#https-proxy) and many others)

Closes #48

:hammer_and_pick: with :heart: by [Siemens](https://opensource.siemens.com/)

### CODE REVIEW

File: src/httpUtils.ts
Comment (fgreinacher) [inline src/httpUtils.ts:1]: I see the `dist` directory is also committed. Should I update it within this PR or will that happen automatically as part of the release workflow?

Comment (davideicardi) [reply src/httpUtils.ts:1]: Yes, thank you!

Comment (fgreinacher) [reply src/httpUtils.ts:1]: Ok, done (`npm run src-build` from a GitHub Codespace)

**Comment by fgreinacher** [comment]
@davideicardi Let me know if something is missing, I'm happy to rework as needed.

**Comment by davideicardi** [comment]
LGTM! Thank you @fgreinacher !
I see that checks are failing, but it is a problem with some old dependencies, not related to your changes.
I have just made a fix on main branch. If you rebase your branch it should work.

**Comment by fgreinacher** [comment]
Thanks @davideicardi! I have just rebased my branch. Ready for another round 🙇

**Comment by davideicardi** [comment]
Thank you!!

**Comment by fgreinacher** [comment]
Thanks for merging @davideicardi, if you could also create a new release that would be awesome 🙇

**Comment by fgreinacher** [comment]
> @fgreinacher Just released v0.19.0!

Thanks a ton :heart:

**Pull request merged**